src/extension/src/ActionHandler.py

```diff
@@ -117,30 +117,33 @@ def setup_telemetry(self):
 
+            self.telemetry_writer.events_folder_path = events_folder
+            # As this is a common function used by all handler actions, setting operation_id such that it will be the same timestamp for all handler actions, which can be used for identifying all events for an operation.
+            # NOTE: Enable handler action will set operation_id to activity_id from config settings. And the same will be used in Core.
+            self.telemetry_writer.set_operation_id(self.operation_id_substitute_for_all_actions_in_telemetry)
+
+            events_folder_previously_existed = True
             if not os.path.exists(events_folder):
                 os.mkdir(events_folder)
                 self.logger.log("Events folder path found in HandlerEnvironment but does not exist on disk. Creating now. [Path={0}][AgentVersion={1}]".format(
                     str(events_folder), str(self.telemetry_writer.get_agent_version())))
+                events_folder_previously_existed = False
 
-            self.telemetry_writer.events_folder_path = events_folder
-            # As this is a common function used by all handler actions, setting operation_id such that it will be the same timestamp for all handler actions, which can be used for identifying all events for an operation.
-            # NOTE: Enable handler action will set operation_id to activity_id from config settings. And the same will be used in Core.
-            self.telemetry_writer.set_operation_id(self.operation_id_substitute_for_all_actions_in_telemetry)
+            self.__log_telemetry_info(telemetry_supported=True, events_folder_previously_existed=events_folder_previously_existed)
         else:
+            # This line only logs to file since events_folder_path is not set in telemetry_writer
             self.__log_telemetry_info(telemetry_supported=False)
 
-    def __log_telemetry_info(self, telemetry_supported):
+    def __log_telemetry_info(self, telemetry_supported, events_folder_previously_existed=False):
         """ Logs detailed information about telemetry and logs an error if telemetry is not supported. """
         events_folder = self.ext_env_handler.events_folder
         events_folder_str = str(events_folder) if events_folder is not None else ""
-        events_folder_exists = os.path.exists(events_folder) if events_folder is not None else False
-        env_var_supports_telemetry = self.telemetry_writer.is_agent_compatible()
-        telemetry_info = "[EventsFolder=\'{0}\'][EventsFolderExists={1}][EnvVar={2}]".format(
-            events_folder_str, str(events_folder_exists), env_var_supports_telemetry)
+        agent_env_var_code = self.telemetry_writer.agent_env_var_code
+        telemetry_info = "[EventsFolder={0}][EventsFolderExistedPreviously={1}][EnvVarCode={2}]".format(
+            events_folder_str, str(events_folder_previously_existed), str(agent_env_var_code))
 
-        if env_var_supports_telemetry is True:
+        if agent_env_var_code == 0:
```

Comment:
This is a prime candidate for an enumeration defined in Constants. The meaning of 0, 1, 2 don't mean anything to someone trying to parse the code quickly, and encountering this in other files.

Reply:
Done.